src/Uno.UI/UI/Xaml/Controls/ToolTip/ToolTip.cs

```diff
@@ -73,10 +129,600 @@ private void AttachToPopup()
+			// is now a FadeIn/FadeOut which doesn't use any FromHorizontalOffset/FromVerticalOffset.  This leaves
+			// ToolTipTemplateSettings and all associated code in a basically non-used and deprecated state - but as of
+			// now, we can't make any breaking changes.  So, to preserve max compatibility, the ToolTipTemplateSettings
+			// is still around as a class/interface/property of the tooltip for now.  It should be removed (or at least
+			// the ToolTip-specific-ness of the Tooltip's template settings should be removed) as soon as it's ok to
+			// make a breaking change.
+		}
+
+
+		private void CalculateTooltipClip(
+			Rect toolTipRect, double maxX, double maxY)
+		{
+			var clipSize = default(Size);
+			var dX = 0.0;
+			var dY = 0.0;
+
+			dX = toolTipRect.Left + toolTipRect.Right - maxX;
+			dY = toolTipRect.Top + toolTipRect.Bottom - maxY;
+
+			if ((dX >= 0) || (dY >= 0))
+			{
+				dX = Math.Max(0, dX);
+				dY = Math.Max(0, dY);
+				clipSize.Width = Math.Max(0, toolTipRect.Right - dX);
+				clipSize.Height = Math.Max(0, toolTipRect.Bottom - dY);
+				PerformClipping(clipSize);
+			}
+		}
+
+		private void MovePointToPointerToolTipShowPosition(
+			ref Point point,
+			PlacementMode placement)
+		{
+			// If the point is inside the placement rect, move it out of the placement rect.
+			var placementRect = GetPlacementRectInWindowCoordinates();
+
+			if (!placementRect.IsEmpty && placementRect.Contains(point))
+			{
+				switch (placement)
+				{
+					case PlacementMode.Left:
+						point.X = placementRect.X;
+						break;
+					case PlacementMode.Right:
+						point.X = placementRect.X + placementRect.Width;
+						break;
+					case PlacementMode.Top:
+					case PlacementMode.Mouse:
+						point.Y = placementRect.Y;
+						break;
+					case PlacementMode.Bottom:
+						point.Y = placementRect.Y + placementRect.Height;
+						break;
+				}
+			}
+		}
+
+		private void MovePointToPointerToolTipShowPosition(
+			ref double left,
+			ref double top,
+			PlacementMode placement)
+		{
+			var point = new Point(left, top);
+
+			MovePointToPointerToolTipShowPosition(ref point, placement);
+
+			left = point.X;
+			top = point.Y;
+		}
+
+		private Rect GetPlacementRectInWindowCoordinates()
+		{
+			var placementRectLocal = Rect.Empty;
+
+			if (PlacementRect is Rect placementRect)
+			{
+				placementRectLocal = placementRect;
+
+				if (Target != null)
+				{
+					var tr = Target.TransformToVisual(null);
+					tr.TransformBounds(placementRectLocal);
+				}
+			}
+
+			return placementRectLocal;
+		}
+
+		// If the owner is a TextElement, we'll get its bounding rect and use that as our placement target rect.
+		// Otherwise, we'll use the default rect derived from the target.
+		private void PerformPlacementInternal()
+		{
+			if (_owner is TextElement textElement)
+			{
+				// UNO TODO
+				//        CCoreServices* pCore = static_cast<CCoreServices*>(DXamlCore::GetCurrent()->GetHandle());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5404644)